commcare_connect/opportunity/tables.py

```diff
@@ -947,3 +947,290 @@ def render_icons(self, record):
+            "index",
+            "user",
+            "suspended",
+            "last_active",
+            "payment_accrued",
+            "total_paid",
+            "last_paid",
+            "total_confirmed_paid",
+        )
+
+
+class WorkerLearnTable(OrgContextTable):
+    index = IndexColumn()
+    user = UserInfoColumn()
+    suspended = SuspendedIndicatorColumn()
+    last_active = tables.DateColumn(format="d-M-Y")
+    started_learning = tables.DateColumn(
+        format="d-M-Y", accessor="date_learn_started", verbose_name="Started Learning"
+    )
+    modules_completed = tables.TemplateColumn(
+        accessor="modules_completed_percentage",
+        template_code="""
+                            {% include "tailwind/components/progressbar/simple-progressbar.html" with text=flag progress=value|default:0 %}
+                        """,
+    )
+    completed_learning = tables.DateColumn(
+        format="d-M-Y", accessor="completed_learn", verbose_name="Completed Learning"
+    )
+    assessment = tables.Column(accessor="passed_assessment")
+    attempts = tables.Column(accessor="assesment_count")
+    learning_hours = DurationColumn()
+    action = tables.TemplateColumn(
+        verbose_name="",
+        orderable=False,
+        template_code="""
+            <div class="opacity-0 group-hover:opacity-100 transition-opacity duration-200 text-end">
+                <a><i class="fa-solid fa-chevron-right text-brand-deep-purple"></i></a>
+            </div>
+        """,
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.use_view_url = True
+        super().__init__(*args, **kwargs)
+
+    class Meta:
+        model = OpportunityAccess
+        fields = ("suspended", "user")
+        sequence = (
+            "index",
+            "user",
+            "suspended",
+            "last_active",
+            "started_learning",
+            "modules_completed",
+            "completed_learning",
+            "assessment",
+            "attempts",
+            "learning_hours",
+            "action",
+        )
+
+    def render_assessment(self, value, record):
+        if not record.date_learn_started:
+            return "--"
+        return "Passed" if value else "Failed"
+
+    def render_action(self, record):
+        url = reverse("opportunity:worker_learn_progress", args=(self.org_slug, record.id))
+        return format_html(
+            """ <div class="opacity-0 group-hover:opacity-100 transition-opacity duration-200 text-end">
+                <a href="{url}"><i class="fa-solid fa-chevron-right text-brand-deep-purple"></i></a>
+            </div>""",
+            url=url,
+        )
+
+
+class WorkerDeliveryTable(tables.Table):
+    use_view_url = True
+
+    id = tables.Column(visible=False)
+    index = IndexColumn()
+    user = tables.Column(orderable=False, verbose_name="Name")
+    suspended = SuspendedIndicatorColumn()
+    last_active = tables.Column()
+    payment_unit = tables.Column()
```

Comment:
Todo: Remove from sortable columns, as it breaks sorting when there are multiple PUs